From 2acbddf23b651615d284bded2ca3c9e525441dce Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 8 Feb 2018 13:30:55 +0100
Subject: [PATCH 13/14] direct3d11: use D3D11_MAP_WRITE to update the shader
 constants

Otherwise the other fields are lost when updating one field
---
 modules/video_output/win32/direct3d11.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 949419f..dbdffd9 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -1048,7 +1048,7 @@ static void UpdateQuadLuminanceScale(vout_display_t *vd, d3d_quad_t *quad, float
     if (quad->f_luminance_scale == luminanceScale)
         return;
 
-    HRESULT hr = ID3D11DeviceContext_Map(sys->d3d_dev.d3dcontext, (ID3D11Resource *)quad->pPixelShaderConstants[0], 0, D3D11_MAP_WRITE_DISCARD, 0, &mappedResource);
+    HRESULT hr = ID3D11DeviceContext_Map(sys->d3d_dev.d3dcontext, (ID3D11Resource *)quad->pPixelShaderConstants[0], 0, D3D11_MAP_WRITE, 0, &mappedResource);
     if (SUCCEEDED(hr)) {
         PS_CONSTANT_BUFFER *dst_data = mappedResource.pData;
         quad->f_luminance_scale = luminanceScale;
@@ -2814,7 +2814,7 @@ static void UpdateQuadOpacity(vout_display_t *vd, const d3d_quad_t *quad, float
     vout_display_sys_t *sys = vd->sys;
     D3D11_MAPPED_SUBRESOURCE mappedResource;
 
-    HRESULT hr = ID3D11DeviceContext_Map(sys->d3d_dev.d3dcontext, (ID3D11Resource *)quad->pPixelShaderConstants[0], 0, D3D11_MAP_WRITE_DISCARD, 0, &mappedResource);
+    HRESULT hr = ID3D11DeviceContext_Map(sys->d3d_dev.d3dcontext, (ID3D11Resource *)quad->pPixelShaderConstants[0], 0, D3D11_MAP_WRITE, 0, &mappedResource);
     if (SUCCEEDED(hr)) {
         PS_CONSTANT_BUFFER *dst_data = mappedResource.pData;
         dst_data->Opacity = opacity;
-- 
2.10.1.windows.1

